### DISCUSSION TITLE
Get HTML element under cursor

### DISCUSSION THREAD

**Discussion by didikunz** [cefsharp/CefSharp.Dom#50] [Q&A]
Hi, is there a way to get the HTML element under the cursor or on a click? 

The idea is to write a data extraction tool, that could be set to automatically read data (capture) from a website. To set things up a user would need to click the items to read data from. Afterwards a automatic script would use that data to select the objects again and read the data anew. - Is that possible?

**Comment by sourabhsharma24** [comment]
> Hi, is there a way to get the HTML element under the cursor or on a click?
> 
> The idea is to write a data extraction tool, that could be set to automatically read data (capture) from a website. To set things up a user would need to click the items to read data from. Afterwards a automatic script would use that data to select the objects again and read the data anew. - Is that possible?
Yes it can be possible ,you can use Mouse Events (Click):
document.addEventListener('click', function(event) {
    var clickedElement = event.target;
 });

**Reply by didikunz** [reply]
If I understand your answer correct this gives me the element under the cursor in JavaScript, right? I would need to get that element from C# or VB.NET, to be able to process it from there. - Thanks anyway.

**Comment by amaitland** [comment]
Yes it's possible. The how would depend on the details. Here are some generic options to get you started.

Option 1:

https://github.com/cefsharp/CefSharp/wiki/Frequently-asked-questions#13-how-do-you-handle-a-javascript-event-in-c
https://github.com/cefsharp/CefSharp/wiki/General-Usage#2-how-do-you-call-a-javascript-method-that-returns-a-result

Option 2:

Excerpt from https://github.com/cefsharp/CefSharp.Dom#dom-access

```c#
await using var devToolsContext = await chromiumWebBrowser.CreateDevToolsContextAsync();

//Event Handler
//Expose a function to javascript, functions persist across navigations
//So only need to do this once
await devToolsContext.ExposeFunctionAsync("jsAlertButtonClick", () =>
{
    _ = devToolsContext.EvaluateExpressionAsync("window.alert('Hello! You invoked window.alert()');");
});

var jsAlertButton = await devToolsContext.QuerySelectorAsync<HtmlButtonElement>("#jsAlertButton");

//Write up the click event listener to call our exposed function
_ = jsAlertButton.AddEventListenerAsync("click", "jsAlertButtonClick");
```

- Expose function
- Add event listener for the specific event you are after

**Reply by didikunz** [reply]
Wow, I had never come up with such a solution myself. Thank you very much.
